include/core/Godot.hpp

```diff
@@ -246,6 +246,12 @@ void register_method(const char *name, M method_ptr, godot_method_rpc_mode rpc_t
 	godot::nativescript_api->godot_nativescript_register_method(godot::_RegisterState::nativescript_handle, ___get_method_class_name(method_ptr), name, attr, method);
 }
 
+// User can specify a derived class D to register the method for, instead of it being inferred.
+template <class D, class B, class R, class... As>
+void register_class_method(const char *name, R (B::*method_ptr)(As...), godot_method_rpc_mode rpc_type = GODOT_METHOD_RPC_MODE_DISABLED) {
+	register_method(name, static_cast<R (D::*)(As...)>(method_ptr), rpc_type);
```

Comment:
This static cast is very unsafe, you can call this function with a method that belongs to a class that does *not* inherit or have a compatible layout with `D`.

Basically if `D` does not inherit `B` then weird things will happen. Is there a way to use templates to check for this inheritance?

Reply:
Originally I had a separate line performing an implicit conversion:
```
R (D::* casted_ptr)(As...) = method_ptr;
```
I didn't notice I had introduced this potential bug when converting it to a one-liner. Thanks for catching it.

I see two obvious solutions:
* Use my original formulation, and the implicit cast won't be possible if D doesn't derive B
* Use `static_assert` with `is_base_of` from `<type_traits>`. Would require including that library, but could provide easier-to-read compile-time error messages.

Reply:
Another possibility is I do both, just to be double safe.

Reply:
Which solution do you prefer?

Reply:
If you can use `static_assert`s with the C++ version we support then I'd say use that.

Since it's in the STL it's not a big deal to have to "include the library", so it should be fine.

Reply:
I forgot to include it but it compiled anyway, so I guess it was included by another header already.